maven-core/src/main/java/org/apache/maven/internal/impl/DefaultLookup.java

```diff
@@ -24,6 +24,8 @@
 
```

Comment:
thinking a bit out loud there but I'm starting to wonder if we should expose guice Injector instead of this container wrapper which basically propagate the status-quo in terms of number of containers in the API and have a single lookup(Type) method.
Probably something to think about before releasing this since ultimately this will need to move from internal to public cause it is a common need to be able to lookup public components programmatically (scripting case being the extreme example).

Reply:
Perso, I'd even move Lookup service to "internal service" (like CoreRealm is), not exposing it to plugins. It serves a very good purpose in Core (to hide Plexus), while plugins can achieve all these with plain JSR330 (or Maven specific JSR330-like annos if they become to existence once).